cli/cmd/install_test.go

```diff
@@ -14,8 +14,10 @@ func TestRender(t *testing.T) {
 		}
 		expectedContent := string(goldenFileBytes)
 
+		testNamespace := "Namespace"
+		controlPlaneNamespace = testNamespace // Set control plane namespace to enforce localhost DNS override.
```

Comment:
I wouldn't expect this test to need to change at all.

Doesn't the setting of `controlPlaneNamespace` need to happen in install.go itself, before calling InjectYAML? In other words, isn't this wallpapering over a real bug by fixing it only during testing but not IRL?

More generally, `InjectYAML` is relying on a bunch of global variables to be set, which are set in the 'RunE` function for the `conduit inject` command, but which may not be set for the `conduit install` command. (This is what's terrible about global variables). Let's make sure that all the global variables that `InjectYAML` depends on are actually set.

Reply:
Consider:

```go
func addInjectInstallSharedFlags(cmd) {
	cmd.PersistentFlags().StringVarP(&conduitVersion, "conduit-version", "v", version.Version, "tag to be used for Conduit images")
	injectCmd.PersistentFlags().StringVar(&initImage, "init-image", "gcr.io/runconduit/proxy-init", "Conduit init container image name")
	injectCmd.PersistentFlags().StringVar(&proxyImage, "proxy-image", "gcr.io/runconduit/proxy", "Conduit proxy container image name")
	injectCmd.PersistentFlags().StringVar(&imagePullPolicy, "image-pull-policy", "IfNotPresent", "Docker image pull policy")
	injectCmd.PersistentFlags().Int64Var(&proxyUID, "proxy-uid", 2102, "Run the proxy under this user ID")
	injectCmd.PersistentFlags().StringVar(&proxyLogLevel, "proxy-log-level", "warn,conduit_proxy=info", "log level for the proxy")
}
```

And then changing `init()` in `inject.go` and `install.go` to call 

`addInjectInstallSharedFlags(&injectCmd)` and  `addInjectInstallSharedFlags(&installCmd)` respectively.

That way one can use `conduit install` to configure the proxy the same way in the controller that it is being configured during `conduit inject` elsewhere, which is needed for our testing.

Reply:
> Doesn't the setting of controlPlaneNamespace need to happen in install.go itself, before calling InjectYAML? In other words, isn't this wallpapering over a real bug by fixing it only during testing but not IRL?

No, the code is correct and I was wrong. I was misremembering how command line flags are dealt with. No change is needed here, other than needing to add something like the `addInjectInstallSharedFlags()` stuff.

Reply:
> add something like the addInjectInstallSharedFlags() stuff.

Is it safe to assume this function will be in `root.go`? It looks like its very similar to `addControlPlaneNetworkingArgs` in `root.go`

Reply:
> Is it safe to assume this function will be in root.go? It looks like its very similar to addControlPlaneNetworkingArgs in root.go

I would probably put it into inject.go so that everybody who adds a new flag to `conduit inject` is made away that they should consider whether it needs to be added to `conduit install` too.